libsolv: 0.7.20 -> 0.7.21

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/libsolv/versions.


meta.description for libsolv is: "A free package dependency solver"

meta.homepage for libsolv is: "https://github.com/openSUSE/libsolv"

meta.changelog for libsolv is: ""


###### Updates performed
- Version update

###### To inspect upstream changes



- [Compare changes on GitHub](https://github.com/openSUSE/libsolv/compare/0.7.20...0.7.21)

###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/rpmdb2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/mergesolv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/dumpsolv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/testsolv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/installcheck had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/repo2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/rpmmd2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/deltainfoxml2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/rpms2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/updateinfoxml2solv had a zero exit code or showed the expected version
- Warning: no invocation of /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin/repomdxml2solv had a zero exit code or showed the expected version
- 0 of 11 passed binary check by having a zero exit code.
- 0 of 11 passed binary check by having the new version present in output.
- found 0.7.21 with grep in /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21
- found 0.7.21 in filename of file in /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
8 total rebuild path(s)

5 package rebuild(s)

5 x86_64-linux rebuild(s)
3 x86_64-darwin rebuild(s)


First fifty rebuilds by attrpath
libdnf
libsolv
microdnf
micromamba
rpm-ostree
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A libsolv https://github.com/r-ryantm/nixpkgs/archive/ae9419f21c1aae79a6f96c8576d7a959ffaf23ad.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21
ls -la /nix/store/99x7pmdljcylmcp44sshmdn4d8l49f8p-libsolv-0.7.21/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>micromamba</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>libdnf</li>
    <li>libsolv</li>
    <li>microdnf</li>
    <li>rpm-ostree</li>
  </ul>
</details>

---

###### Maintainer pings

cc @copumpkin for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

risicle commented:
On nixos x86_64:

```
builder for '/nix/store/vg7dr3zd7c5f2va5sw3x9sydf3xkyhn7-libsolv-0.7.21.drv' failed with exit code 1; last 10 log lines:
  4 out of 4 hunks ignored -- saving rejects to file src/solver.c.rej
  patching file src/solver.h
  Reversed (or previously applied) patch detected!  Assume -R? [n]
  Apply anyway? [n]
  Skipping patch.
  3 out of 3 hunks ignored -- saving rejects to file src/solver.h.rej
  patching file src/solverdebug.c
  patch unexpectedly ends in middle of line
  Hunk #1 FAILED at 130.
  1 out of 1 hunk FAILED -- saving rejects to file src/solverdebug.c.rej
cannot build derivation '/nix/store/hdasjl7z3yjmgj7v642z2gfmhv4jjxh5-micromamba-0.18.1.drv': 1 dependencies couldn't be built
```

looks like `micromamba`'s overridden `libsolv`'s patches won't apply now.

jtojnar committed:
libsolv: 0.7.20 → 0.7.21

The priority patch applied by micromamba has been upstreamed:
https://github.com/openSUSE/libsolv/commit/91be77ea69548808f94339928b5c92f5d9398a54

jtojnar commented:
Should be fixed now.

risicle reviewed (approved):
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64
